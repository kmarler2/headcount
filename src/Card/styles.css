.card {
  height: 350px;
  width: 23.6%;
  display: inline-block;
  text-align: center;
  font-size: 18px;
  margin: 10px;
  background-color: #BDD5EA;
  box-shadow: 3px 3px 0px 0px black;
}

ul {
  padding: 0;
  list-style-type: none;
}

.location {
  color: #577399;
}