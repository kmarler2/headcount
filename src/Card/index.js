import React from 'react';
import './styles.css';

const Card = (props) => {
  return (
    <div 
      className={`card ${props.select ? ' select' : ''}`}
      onClick={()=> { props.findDistrictByClick(props.location); }}
    >
      <h3 className='location'>{props.location}</h3>
      <ul>
        {Object.keys(props.stats).map((stat, index) => {
          return <li 
            key={index}
            style={props.stats[stat] >= 0.5 ? {color: '#F7F7FF'} : {color: '#495867'}}
            >
            {stat}: {props.stats[stat]}
            </li>; 
          })
        }
      </ul>
    </div>
  );
};

export default Card;