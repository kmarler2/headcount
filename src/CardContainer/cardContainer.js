import React from 'react';
import Card from '../Card/Card.js';

const CardContainer = ({data}) => {
  const districtData = Object.keys(data).map((district, index) => {
    return <Card 
    location={data[district].location}
    stats={data[district].stats}
    key={index}
    select={data[district].selected}
    />
  })
  return(
    <div>
      {districtData}
    </div>
    )
}

export default CardContainer;