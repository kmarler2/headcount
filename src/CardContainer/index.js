import React from 'react';
import Card from '../Card/index.js';
import PropTypes from 'prop-types';

const CardContainer = ({data, findDistrictByClick}) => {
  const districtData = Object.keys(data).map((district, index) => {
    return <Card 
      location={data[district].location}
      stats={data[district].stats}
      key={index}
      select={data[district].selected}
      findDistrictByClick={findDistrictByClick}
    />;
  });
  return (
    <div>
      {districtData}
    </div>
  );
};

CardContainer.propTypes = {
  data: PropTypes.object,
  findDistrictByClick: PropTypes.func
};

export default CardContainer;