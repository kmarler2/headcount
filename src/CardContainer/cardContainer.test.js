import React from 'react';
import CardContainer from './index';
import { shallow } from 'enzyme';
import Card from '../Card/index'

describe('CardContainer', () => {
  let wrapper;
  let mockData = {}
  let mockFindDistrictByClick = jest.fn()
  beforeEach(() => {
    wrapper = shallow(<CardContainer
                        data={mockData}
                        findDistrictByClick={mockFindDistrictByClick}
                      />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should create the correct number of cards', () => {

  })
})