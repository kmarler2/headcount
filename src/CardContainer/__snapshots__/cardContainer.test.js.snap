// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    data={Object {}}
    findDistrictByClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
    },
    "ref": null,
    "rendered": Array [],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
